torch/csrc/jit/tensorexpr/bounds_inference.cpp

```diff
@@ -128,36 +192,49 @@ static const Expr* simplifyMax(const Expr* a, const Expr* b) {
  * produce:
  *    [{a, kLoad, 0, 110}, {b, kStore, 0, 100}]
  */
-static BoundsInfo mergeTensorAccesses(const BoundsInfo& unmerged) {
+BoundsInfo mergeTensorAccesses(const BoundsInfo& unmerged) {
   BoundsInfo res;
-  std::unordered_map<const Buf*, TensorAccessBoundsInfo> merged;
-  for (const auto& t : unmerged) {
-    if (!merged.count(t.buf)) {
-      merged[t.buf] = t;
-      continue;
-    }
-
-    // We already have some range for this buf, try to merge them
-    TensorAccessBoundsInfo old_t = merged.at(t.buf);
-    TensorAccessBoundsInfo new_t = t;
-
-    for (size_t i = 0; i < old_t.start.size(); i++) {
-      new_t.start[i] = simplifyMin(old_t.start[i], new_t.start[i]);
-    }
-    for (size_t i = 0; i < old_t.stop.size(); i++) {
-      new_t.stop[i] = simplifyMax(old_t.stop[i], new_t.stop[i]);
+  for (auto& pair : unmerged) {
+    std::vector<TensorAccessBoundsInfo>& res_vec = res[pair.first];
+    for (const auto& v : pair.second) {
+      bool found = false;
+      for (auto& v2 : res_vec) {
+        if (v2.kind != v.kind) {
+          continue;
+        }
+
+        assert(v.start.size() == v.stop.size());
+        assert(v2.start.size() == v2.stop.size());
+        assert(v.start.size() == v2.start.size());
+
+        std::vector<const Expr*> start;
+        std::vector<const Expr*> stop;
+        bool fail = false;
+        for (unsigned int i = 0; i < v.start.size(); ++i) {
```

Comment:
The code is a bit hard to read, maybe use more descriptive names for variables and/or specify types instead of using auto?

Reply:
fair, I'll clean it up a bit.